### PR TITLE
Feature map model

### PR DESCRIPTION
**Pull request by manuandru**
- Concept of Map Model
  - Hexagonal Map
    - Hexagon based on Cubic coordinates
    - A Concept of Map with Hexagonal tiles and knowledge of space
  - Concept of Graph
    - `Node == Spot`
    - `Edge == Road`
  - Add Terrain to the Map
- Utils
  - UnorderedTuples
- Test: add scalatestplus scalacheck to check with properties

### CODE REVIEW

File: .gitignore
Comment (manuandru) [inline .gitignore:1]: Add ignore in order to use viscide

File: build.sbt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ThisBuild / scalaVersion := "3.3.0"
 
 wartremoverWarnings ++= Warts.all
+wartremoverWarnings -= Wart.Equals
```
Comment (manuandru) [inline build.sbt:4]: I think we are old enough of an age to understand the difference between comparison by value and by reference, so I disabled the the warning on ==

File: build.sbt [hunk 2]
```diff
@@ -12,7 +13,9 @@ lazy val scatan = (project in file("."))
       "org.scalatest" %%% "scalatest" % "3.3.0-SNAP4" % Test,
       "org.scalatest" %% "scalatest" % "3.3.0-SNAP4" % Test,
       "org.scala-js" %%% "scalajs-dom" % "2.6.0",
-      "com.raquo" %%% "laminar" % "16.0.0"
+      "com.raquo" %%% "laminar" % "16.0.0",
+      "org.typelevel" %%% "cats-core" % "2.10.0",
+      "org.scalatestplus" %%% "scalacheck-1-17" % "3.2.16.0" % Test
```
Comment (manuandru) [inline build.sbt:18]: I've tried these two libraries

File: src/test/scala/scatan/utils/UnorderedTupleTest.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  "A UnorderedPair" should "have 2 elements" in {
+    assertCompiles("UnorderedPair(0, 0)")
+    assertDoesNotCompile("UnorderedPair()")
+  }
+
+  it should "be equals based on its elements" in {
+    forAll { (a: Int, b: Int) =>
+      val pair1 = UnorderedPair(a, b)
+      val pair2 = UnorderedPair(a, b)
+      pair1 shouldBe pair2
```
Comment (manuandru) [inline src/test/scala/scatan/utils/UnorderedTupleTest.scala:22]: An example of property check
```
    forAll { (a: Int, b: Int) =>
      val pair1 = UnorderedPair(a, b)
      val pair2 = UnorderedPair(a, b)
      pair1 shouldBe pair2
```
e.g. `a` & `b` get some random number during tests

**Review by luigi-borriello00 — APPROVED** [review]
Nice Job

**Review by alemazzo — APPROVED** [review]

**Pull request merged**